lib/discourse_ip_info.rb

```diff
@@ -0,0 +1,27 @@
+require_dependency 'maxminddb'
+
+class DiscourseIpInfo
+
+  def initialize
+    @mmdb_filename = File.join(Rails.root, 'vendor', 'data', 'GeoLite2-City.mmdb')
+
+    begin
+      @mmdb = MaxMindDB.new(@mmdb_filename)
```

Comment:
lets use the low memory reader per:  MaxMindDB::LOW_MEMORY_FILE_READER ... and then if we really want to we can put an LRU cache in front, do a quick bench to see how fast the low mem reader is.

Reply:
We should ship with a small test mmdb file so we can ensure lookups remain good and so on across gem updates. Plus it will allow us to properly test this class

Reply:
```
                               user     system      total        real
Low memory                 4.910000   1.980000   6.890000 (  6.890183)
Low memory (10% cached)    4.890000   2.180000   7.070000 (  7.071249)
Low memory (50% cached)    0.020000   0.010000   0.030000 (  0.022946)
Low memory (100% cached)   0.010000   0.010000   0.020000 (  0.023665)
Low memory (4 threads)    25.520000  24.060000  49.580000 ( 29.471233)
Default                    2.010000   0.000000   2.010000 (  2.011072)
Default (10% cached)       2.120000   0.000000   2.120000 (  2.121219)
Default (50% cached)       0.030000   0.000000   0.030000 (  0.032013)
Default (100% cached)      0.030000   0.000000   0.030000 (  0.031720)
Default (4 threads)        2.070000   0.000000   2.070000 (  2.079891)
```

Testing methodology:
* list of 40 IPs
* the test was looking up the 40 ip 50 times

It looks like the low memory reader with a cache does pretty good, however in my case the IPs repeated themselves quite a lot. Otherwise, the results would have been a lot different.

To be honest, I do not think that the default reader is that bad. 64MBs of data is not a lot (considering the recommended machine has enough memory) and the cache would not do much in a real world situation when IPs happen to more unique.